### ISSUE TITLE
Features JuliaDynamics would like Makie to have

### ISSUE THREAD

**Issue by Datseris** [JuliaPlots/Makie.jl#231]
Hey there, JuliaDynamics is planning to switch everything to Makie once it is stable enough. In parallel we are developing some interactive applications using Makie. This issue is simply a reference list of what is missing before we can fully switch to Makie.

I will be editing the list as I go along and any contributor here is welcome to "check" implemented features. I also hope to contribute myself some of the following, but no promises.

- [ ] Support for the existence of multiple GUIs in parallel. I.e. `scatter` opens a new window instead of replacing the existing window.

- [ ] Selectable zoom (i.e. zooming into a selected rectangle instead of the center of the scene, which is what the mouse wheel does now).

- [ ] a way for the axis to not change when plotting something new, i.e. xlim(), ylim().

- [ ] Markers in dimensionless points.

- [ ] Markers that have the same relative size on the screen when you zoom.

- [ ] to be able to create subplots and to return to them and plot again

- [ ] for subplots that are one below each other, to be able to share the x-axis. And thus when zooming/moving one subplot, all of them move (at least horizontally). This is achieved in PyPlot with: `fig, axs = subplots(D, 1, sharex = true)`

- [ ] Selecting rectangle interactively. `mouse = to_world(scene, Point2f0(mpos))` allows to get a single cliked point, but I would like to get the 2 corners of a selected rectangle (some how click, drag, unclick gives 2 points (the two corners)).

**Comment by SimonDanisch** [comment]
For the last point, that's pretty much what this code does:

**Comment by Datseris** [comment]
Alright, but it will need a docstring and to be exported and to be somewhere in the documentation page :)

**Comment by Datseris** [comment]
@SimonDanisch  how is the selectable zoom achieved? I dont see a way to do it using the mouse.

**Comment by Datseris** [comment]
You are correct, but in interactive plots that clicking mouse does something, this also triggers the click. How shall I modify this: `ispressed(scene, Mouse.left)` to not include it if space is pressed?

**Comment by SebastianM-C** [comment]
I think you could also check for `Keyboard.space` and filter

**Comment by asinghvi17** [comment]
@Datseris I believe the fix for `textslider` (in JuliaPlots/AbstractPlotting#84) has been merged and released.

**Comment by asinghvi17** [comment]
And colorbars are pretty decent now through `colorlegend`

**Comment by Datseris** [comment]
Can you link the doc of colorlegend please ?

**Comment by asinghvi17** [comment]
Welp, it's not up yet but it's in the latest tag of MakieGallery, I guess were just waiting on the website to be rebuilt.  It's in docs/src/colors.md.

**Comment by Datseris** [comment]
I've updated this a bit.

**Comment by asinghvi17** [comment]
Color bars are pretty easy now

**Comment by SimonDanisch** [comment]
Well, not easy enough ;)

**Comment by asinghvi17** [comment]
Oh you mean autogenerating them?

**Comment by SimonDanisch** [comment]
you should just be a flag!

**Comment by asinghvi17** [comment]
Would need better legend support as well.  Is that something you're planning to do during the hackathon, or is there something I can do to help?

**Comment by asinghvi17** [comment]
@Datseris, JuliaPlots/AbstractPlotting.jl#180 fixes:
> a way for the axis limits to not change when plotting something new, i.g. setting xlim(), ylim().

**Comment by Datseris** [comment]
Cool! BTW everyone please just feel free to "tick off" things that are implemented as time goes by
